File: WiMi/WiMi.Domain/ServiceExecutionResult.cs
using System.Collections.Generic;
using WiMi.CrossCutting;

namespace WiMi.Domain
{
    public class ServiceExecutionResult
    {
		List<Error> errors;
        public IReadOnlyList<Error> Errors { get { return errors.AsReadOnly(); } }
        public bool IsOk { get { return errors.IsEmpty(); } }

        public ServiceExecutionResult()
        {
            errors = new List<Error>();
        }

        public ServiceExecutionResult(Error error)
        {
            errors = new List<Error> { error };
        }

		public ServiceExecutionResult(List<Error> errors) : this()
        {
			if (errors is null) { return; }
			this.errors = errors;
        }

        public void AddError(Error error)
        {
            if (error is null) { return; }
            errors.Add(error);
        }

        public void AddErrors(IReadOnlyList<Error> errors)
        {
            if(errors is null) { return; }
            if(errors.IsEmpty()) { return; }
			this.errors.AddRange(errors);
        }
    }
}

File: TuduManayer/java/TuduManager/domain/src/main/java/net/seymourpoler/tudumanager/domain/ServiceExecutionResult.java
package net.seymourpoler.tudumanager.domain;

import java.util.List;

public class ServiceExecutionResult {
    private final List<Error> errors;

    private ServiceExecutionResult(List<Error> errors){
        this.errors = errors;
    }

    private ServiceExecutionResult(){
        this.errors = List.of();
    }

    public Boolean isOk(){
        return errors.isEmpty();
    }

    public List<Error> errors(){
        return errors;
    }

    public static ServiceExecutionResult of(List<Error> errors){
        return new ServiceExecutionResult(errors);
    }

    public static ServiceExecutionResult ok(){
        return new ServiceExecutionResult();
    }
}
